### PR TITLE
refactor: migrate to `dpi` crate

### LINKED ISSUE #1172: Incorrect and incosistent window size with multiwebview
**Describe the bug**

**Steps To Reproduce**
```console
git clone https://github.com/tauri-apps/wry
cd wry
cargo run --example multiwebview
```

**Expected behavior**
The sizes should be of the whole window and same as it is in MacOS

**Screenshots**

## Windows (incorrect size)

image

## MacOS (correct size)

image

**Platform and Versions (please complete the following information):**
```console
OS: Windows 11
Rustc: 
cargo 1.75.0 (1d8b05cdd 2023-11-20)
rustc 1.75.0 (82e1608df 2023-12-21)
```

**Additional context**
It works well on `MacOS`

### LINKED ISSUE #9264: [bug] Wrong Webview size in windows  with 150% scale with tauri 2.0.0-beta.12 unstable feature
### Describe the bug

Build with tauri = { version = "2.0.0-beta.12", features = ["unstable"] }

Display in 4k monitor with 150% scale
fe01cac2c1a9e75126b0439ddc0c568

Display in 1080 monitor with 100% scale works fine
image

### Reproduction

_No response_

### Expected behavior

image

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22631 X64
    ✔ WebView2: 122.0.2365.92
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.77.0 (aedd173a2 2024-03-17)
    ✔ cargo: 1.77.0 (3fe68eabf 2024-02-29)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 16.17.1
    - yarn: 1.22.19
    - npm: 8.15.0

[-] Packages
    - tauri [RUST]: 2.0.0-beta.12
    - tauri-build [RUST]: 2.0.0-beta.10
    - wry [RUST]: 0.37.0
    - tao [RUST]: 0.26.1
    - tauri-cli [RUST]: 1.5.10
    - @tauri-apps/api [NPM]: 2.0.0-beta.4
    - @tauri-apps/cli [NPM]: 2.0.0-beta.6

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://127.0.0.1:1420/
    - framework: Vue.js
    - bundler: Vite
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
closes #1172
closes tauri-apps/tauri#9264

### CODE REVIEW

**Comment by amrbashir** [comment]
@pewsheen could you review the macOS part and make sure it works fine and is correctly using the right type (logical/physical) when retrieving and setting bounds?

**Comment by pewsheen** [comment]
> @pewsheen could you review the macOS part and make sure it works fine and is correctly using the right type (logical/physical) when retrieving and setting bounds?

Looks good on macOS 👍

File: examples/winit.rs [hunk 1]
```diff
@@ -71,19 +70,21 @@ fn main() -> wry::Result<()> {
...
             })
             .unwrap();
         }
         Event::WindowEvent {
           event: WindowEvent::CloseRequested,
           ..
-        } => evl.exit(),
+        } => {
+          dbg!(_webview.bounds().unwrap());
+        }
```
Comment (pewsheen) [inline examples/winit.rs]:
```suggestion
        } => evl.exit(),
```

File: src/webview2/mod.rs [hunk 2]
```diff
@@ -474,8 +471,8 @@ impl InnerWebView {
       controller.SetBounds(RECT {
         left: 0,
         top: 0,
-        right: bounds.x + bounds.width as i32,
-        bottom: bounds.y + bounds.height as i32,
+        right: bounds.0 as i32,
+        bottom: bounds.1 as i32,
```
Comment (pewsheen) [inline src/webview2/mod.rs]: I think it's `i32` already

Comment (amrbashir) [reply src/webview2/mod.rs]: yeah, you're right

Comment (amrbashir) [inline src/webview2/mod.rs]:
```suggestion
        right: bounds.0,
        bottom: bounds.1,
```

**Review by pewsheen — APPROVED** [review]

**Pull request merged**